tests/ui/slice_iter_any.rs

```diff
@@ -0,0 +1,36 @@
+#![warn(clippy::slice_iter_any)]
+
+fn main() {
+    let vec: Vec<u8> = vec![1, 2, 3, 4, 5, 6];
+    let values = &vec[..];
+    let _ = values.iter().any(|&v| v == 4);
+    //~^ ERROR: using `contains()` instead of `iter().any()` on u8/i8 slices is more efficient
+
+    let vec: Vec<u8> = vec![1, 2, 3, 4, 5, 6];
+    let values = &vec[..];
+    let _ = values.contains(&4);
+    // no error, because it uses `contains()`
+
+    let vec: Vec<u32> = vec![1, 2, 3, 4, 5, 6];
+    let values = &vec[..];
+    let _ = values.iter().any(|&v| v == 4);
+    // no error, because it's not a slice of u8/i8
+
+    let values: [u8; 6] = [3, 14, 15, 92, 6, 5];
+    let _ = values.iter().any(|&v| v == 10);
+    // no error, because it's an array
```

Comment:
Wouldn't it be more efficient to lint there, even though this is an array?

Reply:
The optimization doesn't seem to work for arrays. Because of the implementation I mentioned in [this](https://github.com/rust-lang/rust-clippy/pull/13817#discussion_r1884265258).

Reply:
But isn't it clearer anyway to use `.contains()` rather than `.any(==)`?

Reply:
If you are suggesting that clippy should modify this code style, you may be right. I'll try to make the change.

Reply:
However, after thinking about it, what this lint should do is to propose performance improvements for `u8` and `i8` slices, and it seems appropriate to make this as a separate lint. What do you think? If this is a good idea, I'll implement this as a new lint in another PR.

Reply:
Wouldn't a lint which is more efficient for some types (u8/i8) and not less efficient for some others deserve to be in the performance category? And by the way, I see the same 10+ performance boost in `u32` as well.

It looks like a second lint would also cover this one, I'm not sure two lints are needed. I'll let others weigh in.

Reply:
>  I see the same 10+ performance boost in u32 as well

I didn't check about it. Thank you very much. If so, it seems more reasonable to combine them into one as a single lint.

Reply:
In case you don't have one, [here](https://rfc1149.pastes.sh/bench.rs) is the one I used, for testing `func1` and `func2`, the two versions I wanted to compare.